filter.php

```diff
@@ -558,7 +559,7 @@ public function filter($text, array $options = array()) {
             // Open that as a new doc to pull the video node out.
             $tempdom = new DOMDocument('1.0', 'UTF-8');
             @$tempdom->loadHTML($newtext, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
-            $newvideo = $tempdom->getElementsByTagName('video')[0];
+            $newvideo = $tempdom->getElementById('local-smartmedia-wrapper');
```

Comment:
Going along the previous point, perhaps this should be a class selector instead?